### PR TITLE
Docs: update SkyServe docs.

### PR DESCRIPTION
**Pull request by concretevitamin**
* removing `controller_resources`
* some rewording

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below): rendered locally
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

**Comment by MaoZiming** [comment]
This PR https://github.com/skypilot-org/skypilot/pull/2878 will introduce some changes to the doc

File: docs/source/serving/service-yaml-spec.rst [hunk 1]
```diff
@@ -54,23 +54,18 @@ Available fields:
...
-        instance_type: p3.2xlarge
-        disk_size: 256
+    # Besides the `service` section, the rest is a regular SkyPilot task YAML,
+    # which describes each replica:
 
     resources:
-      # Port to run your service (required). This port will be automatically exposed
-      # by SkyServe. You can access your service at http://<endpoint-ip>:<port>.
+      # Port to run your service on each replica (required). This port will be
+      # automatically exposed by SkyServe.
```
Comment (cblmemo) [inline docs/source/serving/service-yaml-spec.rst:66]:
```suggestion
      # automatically exposed to the public internet by SkyServe.
```
not sure if we want to include it; could discuss

File: docs/source/serving/service-yaml-spec.rst [hunk 2]
```diff
@@ -54,23 +54,18 @@ Available fields:
...
       ports: 8080
       # Other resources config...
 
-    # Then comes your SkyPilot task YAML...
+    # The rest of your SkyPilot task YAML...
```
Comment (cblmemo) [inline docs/source/serving/service-yaml-spec.rst:70]: Humm, this sentence looks a little bit weird; it sounds like the previous section is also a SkyPilot task YAML, i.e. some of your SkyPilot task YAML; the rest of your SkyPilot task YAML. Probably change to sth like "The rest section is a SkyPilot task YAML (which contains a workload running on ports xxxx)"

File: docs/source/serving/sky-serve.rst [hunk 3]
```diff
@@ -74,9 +74,14 @@ Use :code:`sky serve status` to check the status of the service:
 
    <div style="height: 20px;"></div>
 
+.. tip::
+
+  Notice that the two replicas are launched in different regions/clouds for the lowest cost and highest GPU availability.
+  This is performed automatically, like a regular ``sky launch``.
```
Comment (cblmemo) [inline docs/source/serving/sky-serve.rst:80]:
```suggestion
  This is performed automatically on all enabled regions/clouds specified in the task YAML, like a regular ``sky launch``.
```
not sure; cloud discuss.

**Review by cblmemo — APPROVED** [review]
LGTM! Left some nits 🫡

File: docs/source/serving/service-yaml-spec.rst (hunk 1 above)
Comment (concretevitamin) [reply docs/source/serving/service-yaml-spec.rst]: Done.

File: docs/source/serving/service-yaml-spec.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/serving/service-yaml-spec.rst]: Good point. Reworded. Wdyt?

File: docs/source/serving/sky-serve.rst (hunk 3 above)
Comment (concretevitamin) [reply docs/source/serving/sky-serve.rst:80]: I think at this point in the tutorial (Quick tour), we haven't said much about the yaml, and we want to be brief here.

Unless you think the current wording is too ambiguous?

**Review by concretevitamin — COMMENTED** [review]
Thanks @cblmemo. PTAL.

File: docs/source/serving/service-yaml-spec.rst (hunk 2 above)
Comment (cblmemo) [reply docs/source/serving/service-yaml-spec.rst]: LGTM!

File: docs/source/serving/sky-serve.rst (hunk 3 above)
Comment (cblmemo) [reply docs/source/serving/sky-serve.rst:80]: Good point! Let's skip it.

**Review by cblmemo — APPROVED** [review]
Looks great to me!

**Comment by concretevitamin** [comment]
> This PR #2878 will introduce some changes to the doc

Sounds good. Hopefully it's minimal merge conflicts.

**Pull request merged**